gitlab/exceptions.py

```diff
@@ -52,7 +52,24 @@ def __str__(self) -> str:
 
 
 class GitlabAuthenticationError(GitlabError):
-    pass
+    def __init__(
+        self,
+        error_message: Union[str, bytes] = "",
+        response_code: Optional[int] = None,
+        response_body: Optional[bytes] = None,
+        auth_type: str = "",
+    ) -> None:
+        super().__init__(
+            error_message=error_message,
+            response_code=response_code,
+            response_body=response_body,
+        )
+        self.auth_type = auth_type
+
+    def __str__(self) -> str:
+        if self.auth_type:
+            return f"{super().__str__()}: authentication_type: {self.auth_type}"
```

Comment:
If we decide to infer the auth type automatically I'd maybe reformulate this a bit. Not sure exactly how, just `authentication_type` sounds like it's a variable name defined/passed somewhere. We can maybe check around how more verbose exceptions do it in cpython or some other libraries.